test/src/test/java/hudson/model/ViewTest.java

```diff
@@ -243,6 +243,24 @@
         assertTrue(xml, xml.contains("<description>two</description>"));
     }
     
+    @Issue("JENKINS-21017")
+    @Test public void doConfigDotXmlReset() throws Exception {
+        ListView view = listView("v");
+        view.description = "one";
+        WebClient wc = j.createWebClient();
+        String xml = wc.goToXml("view/v/config.xml").getWebResponse().getContentAsString();
+        assertThat(xml, containsString("<description>one</description>"));
+        xml = xml.replace("<description>one</description>", "");
+        WebRequest req = new WebRequest(wc.createCrumbedUrl("view/v/config.xml"), HttpMethod.POST);
+        req.setRequestBody(xml);
+        req.setEncodingType(null);
+        wc.getPage(req);
+        assertEquals(null, view.getDescription()); // did not work
+        xml = new XmlFile(Jenkins.XSTREAM, new File(j.jenkins.getRootDir(), "config.xml")).asString();
+        assertThat(xml, not(containsString("<description>"))); // did not work
```

Comment:
Note that `View` and `AbstractItem` have rather different implementations. `View` tries to load in memory, then `save`s to disk on success; `AbstractItem` saves to a temporary file, then loads from it, then renames to the real file on success. Not sure if there is a reason for the discrepancy, but anyway I am testing the effect on both implementations.